### ISSUE TITLE
How do I use the "Take time from current file" option in Import plain text?

### ISSUE THREAD

**Issue by Faxanadus** [SubtitleEdit/subtitleedit#8946]
I don't understand how to format the text file so Subtitle Edit can extract the timestamps.

![image](https://github.com/user-attachments/assets/09faccae-a697-4a62-aec9-62010e9929f1)

Whatever I try, the program just has a - for Start time, End time, and Duration:

![image](https://github.com/user-attachments/assets/5f73db91-8b80-44f1-b801-6771392c5873)

**Comment by darnn** [comment]
I only just tried it myself, so feel free to take this with a grain of salt, but: `Take time from current file` overrides all the other settings. What it does is, if you have a subtitle file open and then use `Import plain text`, if you check `Take time from current file`, it will do as it says, and use the timecodes from the file that's currently open with the text you've imported. Presumably you don't have a file open when you're doing this, or the file you do have is already one with no timecodes. Try this while having a file with timecodes open first.
Conversely, if you uncheck `Take time from current file`, `Generate time codes` goes into effect, with all its specific settings. Then you'll get a file with automatically generated timecodes.

It would also be helpful if you attached a sample of what you're trying to import.

**Comment by bigsealgames** [comment]
Ah I see, I was hoping I could get the time codes from just one file, with the time codes before the start of each line, something like this:

00:01:06.460 --> 00:01:09.983 Hello!  Here is text line 1

00:01:09.983 --> 00:01:20.682 Here's the next line.

00:01:20.682 --> 00:01:27.190 This is the third line

But it's not too much effort just to create another file and reformat it with just the timecodes I guess.

**Comment by Faxanadus** [comment]
Ah I see, I was hoping I could get the time codes from just one file, with the time codes before the start of each line, something like this:

00:01:06.460 --> 00:01:09.983 Hello! Here is text line 1

00:01:09.983 --> 00:01:20.682 Here's the next line.

00:01:20.682 --> 00:01:27.190 This is the third line

But it's not too much effort just to create another file and reformat it with just the timecodes I guess.

**Comment by darnn** [comment]
It looks like what you have there is some strange form of srt file. I think the simplest solution here is, using your text editor of choice which supports regular expressions (I use EditPad), to perform the following search and replace operation:
Replace:
`(\d\d:\d\d:\d\d.\d\d\d --> \d\d:\d\d:\d\d.\d\d\d) `
With:
`\1\r\n`
(Depending on the regex flavor, of course, but I assume you get the idea.)

The result will be this:

`00:01:06.460 --> 00:01:09.983`
`Hello! Here is text line 1`
` `
`00:01:09.983 --> 00:01:20.682`
`Here's the next line.`
` `
`00:01:20.682 --> 00:01:27.190`
`This is the third line`

And you can just save that as a text file and simply open it in Subtitle Edit, without having to import it.

**Comment by bigsealgames** [comment]
That works, thanks!

**Comment by Faxanadus** [comment]
That works, thanks!

**Comment by Vioguy** [comment]
Wouldn't it be nice if it had an option to create timestamps from scratch by using the waveform of the video and comparing it to your finished text? I think only YouTube uses this technology at an advanced level. I need it for copyright material so YouTube is not a choice. I've contacted several companies that use AI and they all say they can't do it.

**Comment by darnn** [comment]
The best you can do for free is use some flavor of Whisper, and then run the results, along with your finished text, through this:
https://github.com/EtienneAb3d/WhisperTimeSync
For your purposes you only need to run the jar file per the instructions there:
https://github.com/EtienneAb3d/WhisperTimeSync/blob/main/distrib/WhisperTimeSync.jar

**Comment by Vioguy** [comment]
> The best you can do for free is use some flavor of Whisper, and then run the results, along with your finished text, through this: https://github.com/EtienneAb3d/WhisperTimeSync For your purposes you only need to run the jar file per the instructions there: https://github.com/EtienneAb3d/WhisperTimeSync/blob/main/distrib/WhisperTimeSync.jar

So I guess there are some options out there but you need to get into the code/python, etc. Frankly, I will need to take some time to familiarize myself with it. Do you request these prompts through the Google Colab workspace?

**Comment by darnn** [comment]
No, no. You can use Whisper in Subtitle Edit and get a file with somewhat accurate timing (or at least better than nothing). That doesn't require interacting with Python directly. Then you need to download the above jar file and run it from the command line as described in this step:
https://github.com/EtienneAb3d/WhisperTimeSync#synchronize
After having prepared the necessary files, which is to say the subtitle file created by Whisper, and a text file with your finished text. If you've done everything correctly, you should get a new file where the finished text corresponds to the timing. You'll still have to go over it, at least in a few places, to make sure it did it properly, but I've had good results with it so far.
Also, you may get an error if the files you're trying to merge are too large (i.e., in my experience, more than 900 subtitles or so). Just split them up into chunks and run the command on each chunk, and later combine them again in Subtitle Edit.

**Comment by Vioguy** [comment]
> No, no. You can use Whisper in Subtitle Edit and get a file with somewhat accurate timing (or at least better than nothing). That doesn't require interacting with Python directly(...)

Thanks for the details. Will give it a shot.
